src/middlewares/roleAuthorization.js

```diff
@@ -1,21 +1,12 @@
 import Responses from '../utils/response';
 
-const roleAuthorization = {
-  checkIsSuperAdministrator(req, res, next) {
-    const { role } = res.locals.user;
-    if (role === 'super_administrator') {
-      next();
-    } else {
-      return Responses.handleError(403, 'insufficient privileges', res);
-    }
-  },
-
-  checkIsManager(req, res, next) {
-    const { role } = res.locals.user;
-    if (role !== 'manager') {
-      return Responses.handleError(403, 'insufficient privileges', res);
-    }
+const authorize = (allowed) => (req, res, next) => {
```

Comment:
I like how you have updated this function, this is much more flexible to use for everyone. Great insight and problem solving here.

Reply:
Nice, At least now this middleware can be reusable